apps/cowswap-frontend/src/legacy/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -190,17 +190,23 @@ export function UnfillableOrdersUpdater(): null {
     verifiedQuotesEnabled,
   ])
 
+  const updatePendingRef = useRef(updatePending)
   useEffect(() => {
+    updatePendingRef.current = updatePending
+  }, [updatePending])
```

Comment:
If we don't care and this won't cause re-renders, and potentially will trigger very often anyway, do we need the useEffect at all for updating the ref?

Reply:
I think so, right? we want to update the `updatePending` function, so next run uses the latest

Or what do you mean?

Reply:
I mean, useEffect is redudant

Reply:
But why?

I don't understand what you mean:

- `updatePending` changes often
- We use `updatePendingRef` to not restart the timer each time the function changes
- We still nee to make `updatePendingRef.current` point to the latest function, so we can execute it when the time is right
- `useRef(updatePending)` will NOT update it each time `updatePending` changes (it assigns the value on the first render, and then ignores)
image.png

This is why I added the:

```
 useEffect(() => {
    updatePendingRef.current = updatePending
  }, [updatePending])
```

So we always keep the latest reference to the `updatePending` fn

Reply:
Okay i understood now what you meant (after we talk :P )